provisioner/aws.go

```diff
@@ -279,17 +280,23 @@ func (a *awsAdapter) waitForStack(ctx context.Context, waitTime time.Duration, s
 		case cloudformation.StackStatusDeleteComplete:
 			return nil
 		case cloudformation.StackStatusCreateFailed:
-			return errCreateFailed
+			err = errCreateFailed
 		case cloudformation.StackStatusDeleteFailed:
-			return errDeleteFailed
+			err = errDeleteFailed
 		case cloudformation.StackStatusRollbackComplete:
-			return errRollbackComplete
+			err = errRollbackComplete
 		case cloudformation.StackStatusRollbackFailed:
-			return errRollbackFailed
+			err = errRollbackFailed
 		case cloudformation.StackStatusUpdateRollbackComplete:
-			return errUpdateRollbackComplete
+			err = errUpdateRollbackComplete
 		case cloudformation.StackStatusUpdateRollbackFailed:
-			return errUpdateRollbackFailed
+			err = errUpdateRollbackFailed
+		}
+		if err != nil {
+			if stack.StackStatusReason != nil {
+				return fmt.Errorf("%s, reason: %s", err, *stack.StackStatusReason)
```

Comment:
I believe the format should be: "%v, reason: %s", because the value of `err` is not a string.

Reply:
the go vet command can detect these problems, may make sense to add it to the build

Reply:
Thanks!